pkg/services/ngalert/state/persister_async.go

```diff
@@ -0,0 +1,62 @@
+package state
+
+import (
+	"context"
+	"time"
+
+	"github.com/benbjohnson/clock"
+	"go.opentelemetry.io/otel/trace"
+
+	"github.com/grafana/grafana/pkg/infra/log"
+)
+
+type AsyncStatePersister struct {
+	log log.Logger
+	// doNotSaveNormalState controls whether eval.Normal state is persisted to the database and returned by get methods.
+	doNotSaveNormalState bool
+	store                InstanceStore
+}
+
+func NewAsyncStatePersisiter(log log.Logger, cfg ManagerCfg) StatePersister {
+	return &AsyncStatePersister{
+		log:                  log,
+		store:                cfg.InstanceStore,
+		doNotSaveNormalState: cfg.DoNotSaveNormalState,
+	}
+}
+
+func (a *AsyncStatePersister) Async(ctx context.Context, ticker *clock.Ticker, cache *cache) {
```

Comment:
The interface slicing feels somewhat off here. The name, sounds like it does a single non-blocking persist in the background, not that it loops forever. I think it stems us from having a mixture of application layers in the two calls:

- `Sync()` is low level, it just does a single write and it's up to the caller to call it
- `Async()` is high level, it not only just writes async, but also runs on a loop cleaning indefinitely

I guess I expected `Async` to just write once, but in the background. Then, the thing calling it on a loop would live above the interface, we'd start it 100% of the time and also make the sync calls 100% of the time...and the right ones would no-op depending on the setting.

This isn't a strong enough comment to block this PR, and can always be improved in the future... so consider this is a nit for now, especially if the sync path will go away.